maths/carmichael_number.py

```diff
@@ -0,0 +1,49 @@
+"""
+== Carmichael Numbers ==
+A number n is said to be a Carmichael number if it satisfies the following modular arithmetic condition:
+
+    power(b, n-1) MOD n = 1,
+    for all b ranging from 1 to n such that b and
+    n are relatively prime, i.e, gcd(b, n) = 1
+
+Examples of Carmichael Numbers: 561, 1105, ...
+https://en.wikipedia.org/wiki/Carmichael_number
+"""
+
+
+def gcd(a, b):
```

Comment:
You can use https://docs.python.org/3/library/math.html#math.gcd if you want.

```suggestion
def gcd(a: int, b: int) -> int:
    """
    >>> gcd(4, 36)
    4
    >>> gcd(4, 38)
    2
    """
```
These are [doctests](https://docs.python.org/3/library/doctest.html) which are **_required_** for all functions.